presto-hive/src/test/java/com/facebook/presto/hive/parquet/write/TestDataWritableWriter.java

```diff
@@ -0,0 +1,378 @@
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.BinaryObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.BooleanObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.ByteObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.DateObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.DoubleObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.FloatObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.HiveCharObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.HiveVarcharObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.IntObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.LongObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.ShortObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.StringObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.TimestampObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.DecimalTypeInfo;
+import parquet.io.api.Binary;
+import parquet.io.api.RecordConsumer;
+import parquet.schema.GroupType;
+import parquet.schema.OriginalType;
+import parquet.schema.Type;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Copy of org.apache.hadoop.hive.ql.io.parquet.write.DataWritableWriter with patch for empty array and empty map
+ * https://issues.apache.org/jira/browse/HIVE-13632
+ */
+public class TestDataWritableWriter
+{
+    private static final Logger log = Logger.get(DataWritableWriter.class);
+    private final RecordConsumer recordConsumer;
+    private final GroupType schema;
+
+    public TestDataWritableWriter(final RecordConsumer recordConsumer, final GroupType schema)
+    {
+        this.recordConsumer = recordConsumer;
+        this.schema = schema;
+    }
+
+    /**
+     * It writes all record values to the Parquet RecordConsumer.
+     *
+     * @param record Contains the record that are going to be written.
+     */
+    public void write(final ParquetHiveRecord record)
+    {
+        if (record != null) {
+            recordConsumer.startMessage();
+            try {
+                writeGroupFields(record.getObject(), record.getObjectInspector(), schema);
+            }
+            catch (RuntimeException e) {
+                String errorMessage = "Parquet record is malformed: " + e.getMessage();
+                log.error(errorMessage, e);
+                throw new RuntimeException(errorMessage, e);
+            }
+            recordConsumer.endMessage();
+        }
+    }
+
+    /**
+     * It writes all the fields contained inside a group to the RecordConsumer.
```

Comment:
`Writes all ...`

Reply:
the same as above, need your opinion regarding updating copy of Hive Parquet writer